phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java

```diff
@@ -648,17 +648,18 @@ private boolean extractTablesAndIndexes(PhoenixConnection conn) {
         }
 
         if (hasViewIndex) {
-            String viewSql = "SELECT DISTINCT TABLE_NAME, TENANT_ID FROM "
-                    + "SYSTEM.CATALOG "
-                    + "WHERE COLUMN_FAMILY = \'" + dataTableFullName + "\' "
+            String viewSql = "SELECT DISTINCT COLUMN_FAMILY, TENANT_ID FROM "
+                    + "SYSTEM.CHILD_LINK "
+                    + "WHERE TABLE_NAME = \'" + tableName + "\' "
                     + (!StringUtil.EMPTY_STRING.equals(schemaName) ? "AND TABLE_SCHEM = \'"
                     + schemaName + "\' " : "")
                     + "AND LINK_TYPE = "
-                    + PTable.LinkType.PHYSICAL_TABLE.getSerializedValue();
+                    + PTable.LinkType.CHILD_TABLE.getSerializedValue();
 
             ResultSet rs = conn.createStatement().executeQuery(viewSql);
             while (rs.next()) {
-                String viewName = rs.getString(1);
+                String viewFullName = rs.getString(1);
```

Comment:
nit: maybe better to name `viewName` here?
I think it's better to have String `viewFullName` = SchemaUtil.getTableName(schemaName, `viewName`), since this is a full name that contains schema and view name.

Reply:
I meant to use getTableNameFromFullName(); thanks for pointing out though. :)